packages/inference/src/lib/validateOutput.ts

```diff
@@ -0,0 +1,158 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+/**
+ * Heavily inspired by zod
+ *
+ * Re-created to avoid extra dependencies
+ */
+
+import { InferenceOutputError } from "./InferenceOutputError";
+
+export const z = {
+	array<T extends { parse: (value: any) => any }>(
```

Comment:
```suggestion
	array<T extends { parse: (value: any) => any, toString?: () => string }>(
```
should we include `toString` here since it was implemented in all types?


Moreover, could you make it helper type like
```ts
type MyType<T> = { parse: (value: any) => <T>, toString?: () => string };
...
export const z = {
	array<T extends MyType<ReturnType<T["parse"]>[]>>(
```

?